diff --git a/docs/.bazelrc b/docs/.bazelrc
index dc6a3f7..9eb0697 100644
--- a/docs/.bazelrc
+++ b/docs/.bazelrc
@@ -1,3 +1,7 @@
-# Bazel configuration flags for rules_foregin_cc_docs
+# Bazel configuration flags for rules_foreign_cc_docs
 
+# https://github.com/bazelbuild/stardoc/issues/112
+common --incompatible_allow_tags_propagation
+
+common --noenable_bzlmod
 build --workspace_status_command=tools/workspace_status.sh
diff --git a/docs/stardoc_repository.bzl b/docs/stardoc_repository.bzl
index 2153aa8..7c63be7 100644
--- a/docs/stardoc_repository.bzl
+++ b/docs/stardoc_repository.bzl
@@ -7,9 +7,9 @@ def stardoc_repository():
     maybe(
         http_archive,
         name = "io_bazel_stardoc",
+        sha256 = "3fd8fec4ddec3c670bd810904e2e33170bedfe12f90adf943508184be458c8bb",
         urls = [
-            "https://github.com/bazelbuild/stardoc/archive/a0f330bcbae44ffc59d50a86a830a661b8d18acc.zip",
+            "https://mirror.bazel.build/github.com/bazelbuild/stardoc/releases/download/0.5.3/stardoc-0.5.3.tar.gz",
+            "https://github.com/bazelbuild/stardoc/releases/download/0.5.3/stardoc-0.5.3.tar.gz",
         ],
-        sha256 = "e12831c6c414325c99325726dd26dabd8ed4c9efa7b4f27b4d1d9594ec7dfc40",
-        strip_prefix = "stardoc-a0f330bcbae44ffc59d50a86a830a661b8d18acc",
     )
diff --git a/docs/tools/workspace_status.sh b/docs/tools/workspace_status.sh
index eaec25a..5ceb3f7 100755
--- a/docs/tools/workspace_status.sh
+++ b/docs/tools/workspace_status.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 set -euo pipefail
 
